packages/shared/src/lib/share.ts

```diff
@@ -30,3 +30,62 @@ export const getTelegramShareLink = (link: string, text: string): string =>
   `https://t.me/share/url?url=${encodeURIComponent(link)}&text=${text}`;
 export const getEmailShareLink = (link: string, subject: string): string =>
   `mailto:?subject=${subject}&body=${encodeURIComponent(link)}`;
+
+interface GetShareLinkParams {
+  provider: ShareProvider;
+  link: string;
+  text?: string;
+}
+
+export const getShareLink = ({
+  provider,
+  link,
+  text,
+}: GetShareLinkParams): string => {
+  switch (provider) {
+    case ShareProvider.WhatsApp:
+      return getWhatsappShareLink(link);
+    case ShareProvider.Twitter:
+      return getTwitterShareLink(link, text ?? '');
+    case ShareProvider.Facebook:
+      return getFacebookShareLink(link);
+    case ShareProvider.Reddit:
+      return getRedditShareLink(link, text ?? '');
+    case ShareProvider.LinkedIn:
+      return getLinkedInShareLink(link);
+    case ShareProvider.Telegram:
+      return getTelegramShareLink(link, text ?? '');
+    case ShareProvider.Email:
+      return getEmailShareLink(link, text ?? '');
+    default:
+      return link;
+  }
+};
+
+export enum ShareCID {
+  Post = 'share_post',
+  Comment = 'share_comment',
+  Profile = 'share_profile',
+}
```

Comment:
We already have `ReferralCampaignKey` which was mapped to `cid` for referral so I think we can reuse it.

Reply:
I switched to that, can you take a look again? 🙏